src/dft/backends/mklcpu/backward.cpp

```diff
@@ -30,77 +30,236 @@
+inline auto expect_config(DescT &desc, const char *message) {
+    dft::detail::config_value actual{ 0 };
+    desc.get_value(Param, &actual);
+    if (actual != Expected) {
+        throw mkl::invalid_argument("DFT", "compute_backward", message);
+    }
+}
+} // namespace detail
 
 //In-place transform
 template <typename descriptor_type, typename data_type>
 ONEMKL_EXPORT void compute_backward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout) {
-    throw mkl::unimplemented("DFT", "compute_backward", "Not implemented for MKLCPU");
+    detail::expect_config<dft::detail::config_param::PLACEMENT, dft::detail::config_value::INPLACE>(
+        desc, "Unexpected value for placement");
+    auto commit_handle = dft::detail::get_commit(desc);
+    detail::check_commit(desc);
+    auto mklcpu_desc = reinterpret_cast<detail::mklcpu_desc_t>(commit_handle->get_handle());
+
+    sycl::queue &cpu_queue{ commit_handle->get_queue() };
+    sycl::buffer<detail::mklcpu_desc_t, 1> mklcpu_desc_buffer {&mklcpu_desc, sycl::range<1>{1}};
+
+    return cpu_queue.submit([&](sycl::handler& cgh){
+        auto desc_acc = mklcpu_desc_buffer.template get_access<sycl::access::mode::read_write>(cgh);
+        auto inout_acc = inout.template get_access<sycl::access::mode::read_write>(cgh);
+        detail::host_task<detail::kernel_name<detail::mklcpu_desc_t>>(cgh, [=](){
+            DftiComputeBackward(*desc_acc.get_pointer(), inout_acc.get_pointer());
+        });
+    }).wait();
 }
 
 //In-place transform, using config_param::COMPLEX_STORAGE=config_value::REAL_REAL data format
 template <typename descriptor_type, typename data_type>
 ONEMKL_EXPORT void compute_backward(descriptor_type &desc, sycl::buffer<data_type, 1> &inout_re,
-                                    sycl::buffer<data_type, 1> &inout_im) {
-    throw mkl::unimplemented("DFT", "compute_backward", "Not implemented for MKLCPU");
+                                   sycl::buffer<data_type, 1> &inout_im) {
+    detail::expect_config<dft::detail::config_param::COMPLEX_STORAGE, dft::detail::config_value::REAL_REAL>(
+        desc, "Unexpected value for complex storage");
+
+    auto commit_handle = dft::detail::get_commit(desc);
+    detail::check_commit(desc);
+    auto mklcpu_desc = reinterpret_cast<detail::mklcpu_desc_t>(commit_handle->get_handle());
+
+    sycl::queue &cpu_queue{ commit_handle->get_queue() };
+    sycl::buffer<detail::mklcpu_desc_t, 1> mklcpu_desc_buffer {&mklcpu_desc, sycl::range<1>{1}};
+
+    return cpu_queue.submit([&](sycl::handler& cgh){
+        auto desc_acc = mklcpu_desc_buffer.template get_access<sycl::access::mode::read_write>(cgh);
+        auto re_acc = inout_re.template get_access<sycl::access::mode::read_write>(cgh);
+        auto im_acc = inout_im.template get_access<sycl::access::mode::read_write>(cgh);
+
+        detail::host_task<detail::kernel_name<detail::mklcpu_desc_t>>(cgh, [=](){
```

Comment:
thanks for pointing out 👍🏽

Reply:
@hjabird changed in f09b61ecd4d1d13c585a4b4f8fa157ffb3bfaa2e, can you recheck ?

Reply:
I think that declaring the class in the template eg.
```c+
detail::host_task<class host_kernel_split_inplace>(cgh, [=]() { ... });
```
is unusual. If this kind of thing is well supported by DPC++ (ie. used in the documentation), then I guess this is OK here (and I'm happy for you to resolve this comment). Otherwise I think its more correct to have:
```c+
class host_kernel_split_inplace;

void myFunc {
 detail::host_task<host_kernel_split_inplace>(cgh, [=]() { ... });
}
```